msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/MFAStates.kt

```diff
@@ -0,0 +1,617 @@
+                    is MFACommandResult.SelectionRequired -> {
+                        MFARequiredResult.SelectionRequired(
+                            nextState = MFARequiredState(
+                                continuationToken = result.continuationToken,
+                                correlationId = result.correlationId,
+                                scopes = scopes,
+                                config = config
+                            ),
+                            authMethods = result.authMethods.toListOfAuthMethods()
+                        )
+                    }
+                    is INativeAuthCommandResult.APIError -> {
+                        Logger.warnWithObject(
+                            TAG,
+                            result.correlationId,
+                            "getAuthMethods() received unexpected result: ",
+                            result
+                        )
+                        MFAError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            errorCodes = result.errorCodes,
+                            exception = result.exception
+                        )
+                    }
+                    is INativeAuthCommandResult.Redirect -> {
+                        MFAError(
+                            errorType = ErrorTypes.BROWSER_REQUIRED,
+                            error = result.error,
+                            errorMessage = result.errorDescription,
+                            correlationId = result.correlationId
+                        )
+                    }
+                }
+            } catch (e: Exception) {
+                MFAError(
+                    errorType = ErrorTypes.CLIENT_EXCEPTION,
+                    errorMessage = "MSAL client exception occurred in getAuthMethods().",
+                    exception = e,
+                    correlationId = correlationId
+                )
+            }
+        }
+    }
+
+    /**
+     * SendChallengeCallback receives the result for sendChallenge() in MFA flows in native authentication.
+     */
+    interface SendChallengeCallback : Callback<MFARequiredResult>
+
+    /**
+     * Sends a challenge to the authentication method; callback variant.
+     * If an authentication method ID was supplied, the server will send a challenge to the specified method. If no ID is supplied,
+     * the server will send the challenge to the user's default auth method.
+     *
+     * @param callback [com.microsoft.identity.nativeauth.statemachine.states.MFARequiredState.SendChallengeCallback] to receive the result on.
+     * @return The result of the send challenge action.
+     */
+    fun sendChallenge(authMethodId: String? = null, callback: SendChallengeCallback) {
```

Comment:
Why not consider using the authMethod instance as a parameter instead of the String id? There don't seem to be significant drawbacks, and the advantages include not having to validate the authMethodId since it's exclusively created by the SDK.